### PR TITLE
Release/v0.17.0

### PR DESCRIPTION
**Pull request by lgalabru**
Addressing a few DX flow  and performance issues + several adjustments following the hord split, along with adding a testing pattern proposal for securing predicates evaluation.

### CODE REVIEW

File: components/chainhook-cli/Cargo.toml [hunk 1]
```diff
@@ -17,8 +17,7 @@ hex = "0.4.3"
 rand = "0.8.5"
 # tikv-client = { git = "https://github.com/tikv/client-rust.git", rev = "8f54e6114227718e256027df2577bbacdf425f86" }
 # raft-proto = { git = "https://github.com/tikv/raft-rs", rev="f73766712a538c2f6eb135b455297ad6c03fc58d", version = "0.7.0"}
-chainhook-sdk = { version = "0.6.0", default-features = false, features = ["zeromq"], path = "../chainhook-sdk" }
-chainhook-types = { version = "1.0.6", path = "../chainhook-types-rs" }
+chainhook-sdk = { version = "0.7.0", default-features = false, features = ["zeromq"], path = "../chainhook-sdk" }
```
Comment (MicaiahReid) [inline components/chainhook-cli/Cargo.toml:20]: We updated `chainhook-sdk` to 0.7.5, should that be reflected here?

File: components/chainhook-sdk/src/chainhooks/mod.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
 pub mod bitcoin;
 pub mod stacks;
 pub mod types;
+
+#[cfg(test)]
+pub mod tests;
```
Comment (MicaiahReid) [inline components/chainhook-sdk/src/chainhooks/mod.rs:6]:
```suggestion
pub mod tests;

```
(the diff editor was complaining that it's missing a newline at the end of the file)

File: components/chainhook-sdk/src/chainhooks/tests/fixtures/mod.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    };
+}
+
+pub fn load_stacks_block_fixture(json_str: &str) -> StacksBlockData {
+    serde_json::from_str(json_str).unwrap()
+}
+
+pub fn get_stacks_testnet_block(block_height: u64) -> &'static StacksBlockData {
+    TESTNET_STACKS_BLOCK_FIXTURES.get(&block_height).unwrap()
+}
```
Comment (MicaiahReid) [inline components/chainhook-sdk/src/chainhooks/tests/fixtures/mod.rs:18]:
```suggestion
}

```
(the diff editor was complaining that it's missing a newline at the end of the file)

File: examples/arkadiko-data-indexing/vault-monitor/app/assets/config/manifest.js [hunk 4]
```diff
@@ -1,4 +1,3 @@
 //= link_tree ../images
 //= link_directory ../stylesheets .css
-//= link_tree ../../javascript .js
-//= link_tree ../../../vendor/javascript .js
+//= link_tree ../../javascript .js
```
Comment (MicaiahReid) [inline examples/arkadiko-data-indexing/vault-monitor/app/assets/config/manifest.js:3]:
```suggestion
//= link_tree ../../javascript .js

```
(the diff editor was complaining that it's missing a newline at the end of the file)

**Review by MicaiahReid — CHANGES_REQUESTED** [review]

File: components/chainhook-cli/Cargo.toml (hunk 1 above)
Comment (lgalabru) [reply components/chainhook-cli/Cargo.toml]: Addressed with latest

**Review by MicaiahReid — APPROVED** [review]

**Pull request merged**